pkg/util/mount/mount_helper_common.go

```diff
@@ -17,74 +17,29 @@ limitations under the License.
 package mount
 
 import (
-	"fmt"
 	"os"
 
 	"k8s.io/klog"
 )
 
 // CleanupMountPoint unmounts the given path and
 // deletes the remaining directory if successful.
-// if extensiveMountPointCheck is true
-// IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
-// IsNotMountPoint is more expensive but properly handles bind mounts within the same fs.
-func CleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointCheck bool) error {
+func CleanupMountPoint(mountPath string, mounter Interface) error {
 	// mounter.ExistsPath cannot be used because for containerized kubelet, we need to check
 	// the path in the kubelet container, not on the host.
 	pathExists, pathErr := PathExists(mountPath)
-	if !pathExists {
+	if !pathExists && pathErr == nil {
 		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", mountPath)
 		return nil
 	}
-	corruptedMnt := IsCorruptedMnt(pathErr)
-	if pathErr != nil && !corruptedMnt {
-		return fmt.Errorf("Error checking path: %v", pathErr)
-	}
-	return doCleanupMountPoint(mountPath, mounter, extensiveMountPointCheck, corruptedMnt)
-}
-
-// doCleanupMountPoint unmounts the given path and
-// deletes the remaining directory if successful.
-// if extensiveMountPointCheck is true
-// IsNotMountPoint will be called instead of IsLikelyNotMountPoint.
-// IsNotMountPoint is more expensive but properly handles bind mounts within the same fs.
-// if corruptedMnt is true, it means that the mountPath is a corrupted mountpoint, and the mount point check
-// will be skipped
-func doCleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointCheck bool, corruptedMnt bool) error {
-	if !corruptedMnt {
-		var notMnt bool
-		var err error
-		if extensiveMountPointCheck {
-			notMnt, err = IsNotMountPoint(mounter, mountPath)
-		} else {
-			notMnt, err = mounter.IsLikelyNotMountPoint(mountPath)
-		}
-
-		if err != nil {
-			return err
-		}
 
-		if notMnt {
-			klog.Warningf("Warning: %q is not a mountpoint, deleting", mountPath)
-			return os.Remove(mountPath)
-		}
-	}
-
-	// Unmount the mount path
-	klog.V(4).Infof("%q is a mountpoint, unmounting", mountPath)
 	if err := mounter.Unmount(mountPath); err != nil {
-		return err
+		klog.Warningf("Warning: Unmount mountPath %s failed with %v", mountPath, err)
 	}
 
-	notMnt, mntErr := mounter.IsLikelyNotMountPoint(mountPath)
-	if mntErr != nil {
-		return mntErr
-	}
-	if notMnt {
-		klog.V(4).Infof("%q is unmounted, deleting the directory", mountPath)
-		return os.Remove(mountPath)
-	}
-	return fmt.Errorf("Failed to unmount path %v", mountPath)
+	//if the path is still a mount point, os.Remove should fail
+	klog.V(4).Infof("%q is unmounted, deleting the directory", mountPath)
+	return os.Remove(mountPath)
```

Comment:
the doc of os.Remove mentioned Remove removes the named file or (empty) directory. So does it mean if path still a mount point, os.Remove will fail?
Just want to confirm it is absolutely safe to call os.Remove even if path is mounted.

Reply:
yes, you should get 'Resource busy' error.